amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -387,7 +389,7 @@ private void handleFlowError(Throwable t, RecordFlow triggeringFlow) {
             // The ack received for this onNext event will be ignored by the publisher as the global flow object should
             // be either null or renewed when the ack's flow identifier is evaluated.
             FanoutRecordsRetrieved response = new FanoutRecordsRetrieved(
-                    ProcessRecordsInput.builder().records(Collections.emptyList()).isAtShardEnd(true).build(), null,
+                    ProcessRecordsInput.builder().records(Collections.emptyList()).isAtShardEnd(true).childShards(new ArrayList<>()).build(), null,
```

Comment:
Isn't this an invalid state "Being at shardEnd" and "having empty childShards". What are the implications of this?

Reply:
Yes, this is an invalid state. This is in the handleFlowError method where we get an ResourceNotFound exception. So I am constructing an invalid ProcessRecordsInput here. This will trigger the ShardConsumer to execute the ShutdownTask with Shard_End reason. But with the empty list of childShards, it will then change the shutdown reason to Lease_Lost. So finally the shardConsumer will shut down in the Lease_Lost mode. Not sure if this is the best way to handle it.